### DISCUSSION TITLE
Bind to Unix socket instead of port

### DISCUSSION THREAD

**Discussion by mohsen1** [vercel/next.js#26424] [Ideas]
# Feature request
Just like Express, I would like to be able to bind to a [Unix socket](https://en.wikipedia.org/wiki/Unix_domain_socket) like `/sockets/mysocket` instead of a host:port setup. 

Looking at the code I see there is an assumption that server will always be bound to a port on a host:

https://github.com/zeit/next.js/blob/18a9c7e371efc4c487f9c3599c3211ce30009d6c/packages/next/cli/next-dev.ts#L56
## Is your feature request related to a problem? Please describe.

Yes, in our internal system we must bind the server to a socket. This is how our server containers connect to Nginx Kubernetes sidecars.

## Describe the solution you'd like

Probably a `--socket` argument to the `next` CLI command that overrides `--host` and `--port`

**Comment by bjoe2k4** [comment]
Would love to see this too. The feature is essentially ready and the added complexity is minor. Please reconsider merging [#20192](https://github.com/vercel/next.js/pull/20192).

**Comment by timkrins** [comment]
Would be a nice to have.
For the meantime, I used `socat` to redirect pipe traffic manually (for a TCP listener on 127.0.0.1 port 3000):
```
socat UNIX-LISTEN:/var/tmp/localhost_three_thousand,fork TCP:127.0.0.1:3000
```

**Comment by fredericnguyenduy** [comment]
I also would like to have that feature: that would make application management easier when you have multiple nodejs applications running on the same server, each one using a different port, and also when your application uses different ports on different environment... If instead, the application uses the same UNIX socket on all environments, that would make port management easier.
